lib/puppetserver/ca/puppet_config.rb

```diff
@@ -75,19 +74,28 @@ def resolve_settings(overrides = {})
         cadir = overrides.fetch(:cadir, '$ssldir/ca')
         settings[:cadir] = substitutions['$cadir'] = cadir.sub(unresolved_setting, substitutions)
 
-        settings[:cacert] = overrides.fetch(:cacert, '$cadir/ca_crt.pem')
-        settings[:cakey] = overrides.fetch(:cakey, '$cadir/ca_key.pem')
-        settings[:cacrl] = overrides.fetch(:cacrl, '$cadir/ca_crl.pem')
-        settings[:serial] = overrides.fetch(:serial, '$cadir/serial')
+        # Note the call to hostname below, (tbd from Justin's code)
+        settings[:certname] = substitutions['$certname'] = overrides.fetch(:certname, `hostname`.chomp)
+
+        settings[:ca_name] =        overrides.fetch(:ca_name, 'Puppet CA: $certname')
+        settings[:root_ca_name] =   overrides.fetch(:root_ca_name, 'Puppet Root CA: $certname')
+        settings[:ca_ttl] =         overrides.fetch(:ca_ttl, '15y')
+        settings[:keylength] =      overrides.fetch(:keylength, 4096).to_s
```

Comment:
Hrm... We make this a string, I assume, because the value from the puppet.conf will be read as a string and the substitution code below will barf if you call it on a number. But this means the consumer of this object needs to coerce it into a number before using it... What if we coerced this to a number here and then skipped non-string values in the substitution code below? (We could also then remove the call `to_i` in `GenerateAction#to_i`)

iow, I think it would be better for this to return usable settings and not have the consumer care about coercing.